cmd/telegraf/telegraf.go

```diff
@@ -282,6 +282,13 @@ func reloadLoop(stop chan struct{}, s service.Service) {
 		}
 
 		ag.Run(shutdown)
+
+		if *fPidfile != "" {
+			err := os.Remove(*fPidfile)
+			if err != nil {
+				log.Fatalf("E! Unable to remove pidfile: %s", err)
```

Comment:
Does this really need to be fatal? We should already be shutting down, so we don't need to exit early and abort the shutdown process.
And what if the issue is that the file was already removed? We're were just going to remove it anyway.